### PR TITLE
eth: Add tests of Beacon API response structs & update Beacon API client

### PR DESCRIPTION
**Pull request by sebastianst**
**Description**

* Fixes beacon L1 client to use [updated format](https://github.com/ethereum/beacon-APIs/pull/369) for path `eth/v1/beacon/blob_sidecars`
* Adds tests for JSON API response types

**Metadata**

- Fixes https://github.com/ethereum-optimism/protocol-quest/issues/94

### CODE REVIEW

File: op-service/eth/blobs_api.go [hunk 1]
```diff
@@ -9,8 +9,43 @@ type BlobSidecar struct {
...
+	KZGProof          Bytes48                 `json:"kzg_proof"`
+	SignedBlockHeader SignedBeaconBlockHeader `json:"signed_block_header"`
+
+	// TODO(Seb): we don't use this yet afaik
+	// KZGCommitmentInclusionProof [17]Bytes32             `json:"kzg_commitment_inclusion_proof"`
+}
+
+func (sc *APIBlobSidecar) BlobSidecar() *BlobSidecar {
+	return &BlobSidecar{
+		BlockRoot:     sc.SignedBlockHeader.Message.StateRoot, // TODO confirm
```
Comment (roberto-bayardo) [inline op-service/eth/blobs_api.go]: You can remove BlockRoot from the Sidecar struct, it's no longer part of it.

File: op-service/eth/blobs_api.go [hunk 2]
```diff
@@ -1,16 +1,49 @@
...
+		Blob:          sc.Blob,
+		Index:         sc.Index,
+		KZGCommitment: sc.KZGCommitment,
+		KZGProof:      sc.KZGProof,
+	}
+}
+
+type SignedBeaconBlockHeader struct {
+	Message   BeaconBlockHeader `json:"message"`
+	Signature Bytes96           `json:"signature"`
```
Comment (sebastianst) [inline op-service/eth/blobs_api.go]: We don't use this field yet, so we might also just remove it and then also don't need to add `Bytes96`.

File: op-service/eth/types.go [hunk 3]
```diff
@@ -76,6 +74,30 @@ func (b Bytes32) TerminalString() string {
...
+}
+
+func (b Bytes96) String() string {
+	return hexutil.Encode(b[:])
+}
+
+// TerminalString implements log.TerminalStringer, formatting a string for console
+// output during logging.
+func (b Bytes96) TerminalString() string {
+	return fmt.Sprintf("%x..%x", b[:3], b[29:])
```
Comment (protolambda) [inline op-service/eth/types.go]:
```suggestion
	return fmt.Sprintf("%x..%x", b[:3], b[93:])
```

Comment (sebastianst) [reply op-service/eth/types.go]: good catch

**Pull request merged**